tests/e2e-tests/config/fixture-loaders.js

```diff
@@ -0,0 +1,297 @@
+				value: 'specific',
+			},
+			{
+				id: 'woocommerce_specific_ship_to_countries',
+				value: [ 'DZ', 'CA', 'NZ', 'ES', 'GB', 'US' ],
+			},
+			{
+				id: 'woocommerce_enable_coupons',
+				value: 'yes',
+			},
+			{
+				id: 'woocommerce_calc_taxes',
+				value: 'yes',
+			},
+			{
+				id: 'woocommerce_currency',
+				value: 'USD',
+			},
+		],
+	} ).catch( ( error ) => console.log( error ) );
+};
+
+const createTaxes = () => {
+	return WooCommerce.post( 'taxes/batch', {
+		create: [
+			{
+				country: 'US',
+				rate: '5.0000',
+				name: 'State Tax',
+				shipping: false,
+				priority: 1,
+			},
+			{
+				country: 'US',
+				rate: '10.000',
+				name: 'Sale Tax',
+				shipping: false,
+				priority: 2,
+			},
+			{
+				country: 'UK',
+				rate: '20.000',
+				name: 'VAT',
+				shipping: false,
+			},
+		],
+	} )
+		.then( ( response ) =>
+			response.data.create.map( ( taxes ) => taxes.id )
+		)
+		.catch( ( error ) => console.log( error ) );
+};
+
+const deleteTaxes = ( ids ) => {
+	WooCommerce.post( 'taxes/batch', {
+		delete: ids,
+	} ).catch( ( error ) => console.log( error ) );
+};
+const createCoupons = () => {
+	const coupons = [
+		{
+			code: 'coupon',
+			discount_type: 'fixed_cart',
+			amount: '5',
+		},
+		{
+			code: 'oldcoupon',
+			discount_type: 'fixed_cart',
+			amount: '5',
+			date_expires: '2020-01-01',
+		},
+		{
+			code: 'below100',
+			discount_type: 'percent',
+			amount: '20',
+			maximum_amount: '100.00',
+		},
+		{
+			code: 'above50',
+			discount_type: 'percent',
+			amount: '20',
+			minimum_amount: '50.00',
+		},
+		{
+			code: 'a12s',
+			discount_type: 'percent',
+			amount: '100',
+			individual_use: true,
+			email_restrictions: '*@automattic.com%2C *@a8c.com',
+		},
+		{
+			code: 'freeshipping',
+			discount_type: 'percent',
+			amount: '0',
+			free_shipping: true,
+		},
+	];
+	return WooCommerce.post( 'coupons/batch', { create: coupons } )
+		.then( ( response ) =>
+			response.data.create.map( ( coupon ) => coupon.id )
+		)
+		.catch( ( e ) => console.log( e.response.data.message ) );
+};
+const deleteCoupons = ( ids ) => {
+	return WooCommerce.post( 'coupons/batch', {
+		delete: ids,
+	} ).catch( ( error ) => console.log( error ) );
+};
+// currently this only creates a single product for the sake of review
+// more products will be added later
+const createProducts = () => {
+	return WooCommerce.post( 'products/batch', {
+		create: [
+			{
+				name: 'Woo Single #1',
+				type: 'simple',
+				regular_price: '21.99',
+				virtual: true,
+				downloadable: true,
+				downloads: [
+					{
+						name: 'Woo Single',
+						file:
+							'http://demo.woothemes.com/woocommerce/wp-content/uploads/sites/56/2013/06/cd_4_angle.jpg',
+					},
+				],
+				images: [
+					{
+						src:
+							'http://demo.woothemes.com/woocommerce/wp-content/uploads/sites/56/2013/06/cd_4_angle.jpg',
+					},
+				],
+			},
+		],
+	} )
+		.then( ( products ) => {
+			createReviews( products.data.create[ 0 ].id );
```

Comment:
since reviews depend on products, we create them inside of the product.

Reply:
the problem is that for your test to work right, you need to know what product has the reviews exactly, all the data is static but the IDs are not static, so we will still need a way to get the id from the product and pass it to reviews, it could be like this in the setup.js file

```
const productsIds = await createProducts();
await createReviews( productIds[ 0 ] );
```

This coupling will get more apparent once we put Categories and Tags in it since you can only assign a category by its ID in the product.

another approach is to query all products inside the review, filter to the product you want, get its ID and pass it to reviews, a similar thing can happen inside Product and Categories.
This will be an extra request but that isn’t an issue when it comes to tests, however, we need to ensure the order of execution, so we will move away from Promise.all to async-await.